src/Security/PetEditVoter.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace App\Security;
+
+use App\Entity\Pet;
+use App\Entity\User;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Authorization\Voter\Voter;
+
+class PetEditVoter extends Voter
+{
+    protected function supports($action, $subject)
+    {
+        return $action === 'edit' && $subject instanceof Pet;
+    }
+
+    protected function voteOnAttribute($action, $subject, TokenInterface $token)
+    {
+        $user = $token->getUser();
+
+        if (!$user instanceof User) {
+            // the user must be logged in; if not, deny access
+            return false;
+        }
+
+        return $this->canEdit($subject, $user);
+    }
+
+    private function canEdit(Pet $pet, User $user)
+    {
+        return ($user->getId() == $pet->getOwner()->getId() or $user->hasRole("ROLE_ADMIN"));
```

Comment:
never use `or`, ever never ever http://php.net/manual/en/language.operators.logical.php#80005

Reply:
Also please try to always use strict comparison `===` instead of `==`